azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -80,6 +80,64 @@ def _execute_op(self, op):
                 ),
             )
 
+        elif (
+            isinstance(op, pipeline_ops_base.UpdateSasTokenOperation)
+            and self.pipeline_root.connected
+        ):
+            logger.info(
+                "{}({}): Connected.  Passing op down and reconnecting after token is updated.".format(
+                    self.name, op.name
+                )
+            )
+
+            # make a callback that can call the user's callback after the reconnect is complete
+            def on_reconnect_complete(reconnect_op):
```

Comment:
>def on_reconnect_complete(reconnect_op): [](start = 12, length = 40)

This could probably be a private function defined within on_token_update_complete, since it's only used there. Might help readability.